SourceryRuntime/Sources/Common/Composer/ParserResultsComposed.swift

```diff
@@ -51,6 +53,11 @@ internal struct ParserResultsComposed {
             alias.type = resolveType(typeName: alias.typeName, containingType: alias.parent)
         }
 
+        /// Map associated types
+        associatedTypes.forEach {
+            typeMap[$0.key] = $0.value.type
+        }
```

Comment:
@art-divin i'm digging into a regression that I have run into and I want to confirm if this change is the culprit or not...

I have the following example:

```swift
protocol FooValue {}

protocol Foo {
    associatedtype Bar: FooValue
    var value: Bar { get }
}

class Baz {
    struct Bar {
    }

    let bar: Bar

    init(bar: Bar) {
        self.bar = bar
    }
}
```

For the `Bar` associated type inside the `Foo` protocol, this change causes `typeMap["Bar"]` to be set to `FooValue`. 

When I dump the parsed AST for `Baz`, I get expected values from the type of the stored variable, but I get the wrong type from the initialiser parameter:

```
(lldb) po result.first!.storedVariables.first!.typeName!.name
"Baz.Bar"
(lldb) po result.first!.initializers.first!.parameters.last!.type!.name
"FooValue"
```

I have a hunch that the issue comes in two parts:

1. The initializer parameter type isn't getting the global name assigned to its `typeName` the same way that the store variable does:
    ```
    (lldb) po result.first!.storedVariables.first!.typeName
    Baz.Bar
    (lldb) po result.first!.initializers.first!.parameters.last!.typeName
    Bar
2. Associated types are being treated as globals which is a potential cause for conflict

I can try to dig further, or put this into an issue, but I wanted to check to see if this is something that you're actively aware of atm? 

Thanks!

Reply:
Hello @liamnichols ,

amazing work! Very nice of you sharing this 🤝 

> I have a hunch that the issue comes in two parts:

Sounds like it, there were and still are a number of features related to associated types which are not supported at all.
Previously they were completely ignored when mocks were being generated during to missing metadata.

> I can try to dig further, or put this into an issue, but I wanted to check to see if this is something that you're actively aware of atm?

If you can, sure, that would be awesome -> creating an issue, digging further... Sounds like a plan! 
To answer your question, I am not aware of this regression and I really appreciate the code sample of things not working as expected in your comment here.

If you would be able to at least provide "Expected" and "Actual" sections in your issue, say, if you cannot fix this at your convenience, 
it would play a crucial role in speeding up the investigation and the fix in future. 🙏